sympy/ntheory/tests/test_totient_hypothesis.py

```diff
@@ -0,0 +1,15 @@
+from hypothesis import given
+from hypothesis import strategies as st
+from sympy.ntheory import totient
+from sympy import divisors
+import random
+
+
+@given(n=st.integers(1,10**10))
+def test_totient_hypothesis(n):
+    assert totient(n) <= n
+    div = divisors(n)
+    totients = [totient(i) for i in div]
+    assert n == sum(totients)
+    d = random.choice(div)
```

Comment:
You should avoid using `random` in hypothesis tests. This will make the test inherently flaky, because a failure would depend on the random seed. Instead, you should try to use hypothesis itself to do this, so that it can be done reproducibly. 

I *think* the proper way to do this is to use [data](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.data) with [sampled_from](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.sampled_from), but @honno should maybe confirm that.

Reply:
Yeah if the top of the test looked like

```python
@given(n=st.integers(1,10**10), data=st.data())
def test_totient_hypothesis(n, data):
    ...
```

you could write

```suggestion
    d = data.draw(div)
```

Reply:
You could also just check every divisor, though maybe that would be too slow?

Reply:
I think that would be too slow, and not super illuminating. In any case the particular use case for `d`: checking `assert totient(n // (n // d)) == totient(d)` holds was deemed not useful.